src/http_utils.cpp

```diff
@@ -216,6 +218,22 @@ std::string http_utils::standardize_url(const std::string& url) {
     return result;
 }
 
+std::string http_utils::generate_random_upload_filename(std::string &directory) {
+    char *filename = NULL;
+    if (asprintf(&filename, "%s/%s", directory.c_str(), http_utils::upload_filename_template) == -1) {
+        return "";
+    }
```

Comment:
`asprintf()` is a GNU extension, not in C or POSIX. Not sure if this is okay in libhttpserver. 

Also: just returning an empty string on error is somewhat surprising to me. Did not check users of this method. Declaration has no documentation on that.

Reply:
I could not figure out how to use only strings inside this function, as mkstemp needs a char * and actually changes the content of the parameter.
So I created the template filename as a string and strdup-ed this into a char * which is then given to mkstemp (or _mktemp_s on windows).
Afterwards the changed char * is put into an output string which is then returned.

So at least the concatenation of the directory, the path separator and the template is done with standard string methods.

I hope this is acceptable.

The empty return is also exchanged with a `throw std::exception` (and do a try - catch in the calling function).
Actually I do not really have an idea, which exception to use, so I went for a quite general one.
If you have any suggestion for a better exception, please give me a hint

Reply:
No issues with the string.

For the exception, let's create a custom exception so that we can then catch that (and we don't incur in the risk of catching exceptions raised by other parts of the code)